Create CSS Entries for translateX & translateY and Update translate (#576)
Codecademy/docs: content/css/concepts/transform-functions/terms/translate/translate.md

  - 'paths/full-stack-engineer-career-path'
---

> The `translate` function translates an element along the X and/or Y axes.

Reviewer: ```suggestion
The `translate` function moves an element along the X- and/or Y-axis.
```

## Syntax

> 

Reviewer 2: ```suggestion

The required `<X-value>` or `<Y-value>` can be one of the following:

- Length value: `100px`, `1.5em`
- Percentage value: `25%`, `50%`

```

```pseudo

> /* With a single value */

Reviewer 2: ```suggestion
/* Single value */
```

transform: translate(<X-value>);

> /* With both values */

Reviewer 2: ```suggestion
/* Double values */
```

transform: translate(<X-value>, <Y-value>);
```

> where a required `<value>` can be one of the following:

Reviewer: ```suggestion
A required `<X-value>` or `<Y-value` can be one of the following:
```

> A required `<X-value>` or `<Y-value>` can be one of the following:
> 
> - Length value: `100px`, `1.5em`
> - Percentage value: `25%`, `50%`

Reviewer 2: ```suggestion
```
I recommend moving this above the snippet.

> Providing only a single value will represent the horizontal axis. Providing a second comma separated value will represent the vertical axis. A positive value moves the element right along the horizontal axis and down the vertical axis while a negative value moves the element left along the horizontal axis and up the vertical axis.

Reviewer: ```suggestion
Providing only a single value will represent the horizontal axis. Providing a second comma-separated value will represent the vertical axis. A positive value moves the element towards the right along the horizontal X-axis and down the vertical Y-axis. A negative value moves the element left along the horizontal X-axis and up the vertical Y-axis.
```

> Providing only a single value will represent the horizontal axis. Providing a second comma-separated value will represent the vertical axis. A positive value moves the element towards the right along the horizontal X-axis and down the vertical Y-axis. A negative value moves the element left along the horizontal X-axis and up the vertical Y-axis.

Reviewer 2: ```suggestion
A single value will represent only the horizontal axis. Providing a second comma-separated value will represent the vertical axis, as well. As as the direction goes:

* A positive value moves the element towards the right along the horizontal X-axis and down the vertical Y-axis (southeast). 
* A negative value moves the element left along the horizontal X-axis and up the vertical Y-axis (northwest).
```

Reviewer 2: Is there more to this with respect to moving in a northeast or southwest direction?

## Example 1

> Move the `.box` element `100px` right along the horizontal axis:

Reviewer: ```suggestion
Move the element, with a `class` of `box`, `100px` right along the horizontal X-axis:
```

> Move the element, with a `class` of `box`, `100px` right along the horizontal X-axis:

Reviewer 2: ```suggestion
In the example below, an element with a `.box` class is moved `100px` to the right, along the horizontal X-axis:
```

  - 'paths/full-stack-engineer-career-path'
---

The `translate` function moves an element along the X- and/or Y-axis.

## Syntax

The required `<X-value>` or `<Y-value>` can be one of the following:

- Length value: `100px`, `1.5em`
- Percentage value: `25%`, `50%`

```pseudo
/* Single value */
transform: translate(<X-value>);

/* Double values */
transform: translate(<X-value>, <Y-value>);
```

> A single value will represent only the horizontal axis. Providing a second comma-separated value will represent the vertical axis, as well. As as the direction goes:

Reviewer 2: ```suggestion
A single value will represent only the horizontal axis. Providing a second comma-separated value will represent the vertical axis, as well. As far as the direction goes:
```

## Example 2

> Move the `.box` element `200px` tight along the horizontal axis and `100px` up the vertical axis:

Reviewer: ```suggestion
Move the `.box` element `200px` along the horizontal X-axis and `100px` up the vertical Y-axis:
```

> Move the `.box` element `200px` right along the horizontal X-axis and `100px` up the vertical Y-axis:

Reviewer 2: ```suggestion
In this example, the `.box` class element is moved `200px` to the right, along the X-axis, and `100px` up the vertical Y-axis.
```
